### DISCUSSION TITLE
Parameter `index_name_prefix`

### DISCUSSION THREAD

**Discussion by dhirtzbruch** [schranz-search/schranz-search#319] [Q&A] [answered]
Can someone tell me what the `index_name_prefix` in the SymfonyBundle is used for? By the parameter name I thought it should be possible to add a prefix to any index used when creating indexes as well as when querying the data - but I could not get that to work.

Use case would be using the same ElasticSearch cluster for staging / prod environments, and just automatically having the prefix added through the bundle configuration.

**Comment by alexander-schranz** [comment, marked as answer]
It was created for that usecase, specially running tests in another env with a prefix. But if I look there is currently the prefix only used when there are multiple files. So think there is a bug here: https://github.com/schranz-search/schranz-search/blob/efa54fa69471f0afdb81740689afbe40f9f98865/packages/seal/src/Schema/Loader/PhpFileLoader.php#L65-L67

Which misses maybe something like this:

```php
                } else {
                    $index = new Index($this->indexNamePrefix . $name, $index->fields);
                }
```

**Reply by dhirtzbruch** [reply]
Thanks for the reply. Would you mind if I prepare a PR here?

**Reply by alexander-schranz** [reply]
Sure do it 👍

**Reply by dhirtzbruch** [reply]
Just did some tests, your code snippet works when creating indexes, when dropping them and when indexing documents, so I think that should already be enough.

**Reply by dhirtzbruch** [reply]
Should be fixed in #320 / #321
